### ISSUE TITLE
Logic error with registering for buckets when having duplicate fluids from different mods

### ISSUE THREAD

**Issue by HenryLoenwind** [MinecraftForge/MinecraftForge#4863]
"If you have a fluid you don't need to be bucketable, and another mod has the same fluid that needs to be bucketable, YOU must also make buckets for your fluids or the OTHER mod will fail."

The bucket registration is for specific fluid objects, not fluid names. However, mods have to rely that a specific fluid name has a bucket (e.g. when their own fluid loses the election) and the bucket stores the fluid name in its nbt, only. Therefore the bucket fluid list should be fluid names, not fluid objects.

As a side effect of the current implementation, the getter for the bucket fluids will contain many un-identical duplicates (different fluid objects for the same fluid).

TL;DR: `net.minecraftforge.fluids.FluidRegistry.bucketFluids` should be `Set<String>`, not `Set<Fluid>`.
